lib/webtask.js

```diff
@@ -708,6 +756,66 @@ exports.ensure_authenticated = function(ctx) {
             });
         }
 
+        function validate_auth_token(auth) {
+            var auth_token;
+            var decoded_auth_token;
+            var userinfo;
+            var aud_should_be;
+            var secret = auth.secret || ctx.config.default_auth0_client_secret.current;
+
+            if(auth.account)
+                aud_should_be = auth.account.client_id;
+            else
+                aud_should_be = req.headers.host + urlParse(req.url).pathname;
+
+            try {
+                secret = new Buffer(secret, 'base64');
+            } catch(e) {
+                throw { code: 403, message: 'JWT_SECRET must be base64 encoded: ' + e.message, req_id: req.req_id };
+            }
+
+            try {
+                auth_token = (req.query.key || req.headers['authorization']
+                    .match(/^bearer\s+(.+)$/i)[1]);
+            } catch(e) {
+                throw { code: 400, message: 'could not parse authorization header' + e.message, req_id: req.req_id };
+            }
+
+            try {
+                userinfo = jsonwebtoken.verify(auth_token, secret);
+            } catch(e) {
+                throw { code: 401, message: 'invalid authorization token signature', req_id: req.req_id };
+            }
+
+            if(!userinfo.aud)
+                throw { code: 403, message: 'no authorization token \'aud\' claim', req_id: req.req_id };
+
+            if(userinfo.aud !== aud_should_be)
+                throw { code: 403, message: 'invalid authorization token \'aud\' claim', req_id: req.req_id };
+
+            if(auth.email) {
+                if(userinfo.email_verified === false) {
+                    throw { code: 401, message: 'this webtask can only be accessed if your' + get_provider(userinfo.sub) + 'email is verified', req_id: req.req_id };
+                }
+
+                if(auth.email instanceof Array) {
+                    for(var i = 0; i < auth.email.length; i++)
+                        if(userinfo.email.match(auth.email[i])) {
+                            return userinfo;
+                        }
+                }
+                else {
+                    if(userinfo.email.match(auth.email)) {
+                        return userinfo;
+                    }
+                }
+
+                throw { code: 401, message: 'your email is does not authorize you to view this webtask', req_id: req.req_id };
+            } else {
+                return userinfo;
+            }
+        }
```

Comment:
Once we are done with *authentication* of the JWT, we should re-expose all claims of that JWT (whatever they might be) through some property of the `ctx` in the auth0-sandbox, e.g. `ctx.user`. I see you are assigning userinfo to jwt.payload.auth.user. Is this wired all the way through `ctx.user`?